test/object-store/sync/flx_sync.cpp

```diff
@@ -4995,6 +4967,466 @@ TEST_CASE("flx: nested collections in mixed", "[sync][flx][baas]") {
     CHECK(nested_list.get_any(1) == "foo");
 }
 
+TEST_CASE("flx: role change bootstrap", "[sync][flx][baas][role_change][bootstrap]") {
+    const Schema g_person_schema{{"Person",
+                                  {{"_id", PropertyType::ObjectId, Property::IsPrimary{true}},
+                                   {"role", PropertyType::String},
+                                   {"firstName", PropertyType::String},
+                                   {"lastName", PropertyType::String}}}};
+
+    auto fill_person_schema = [](SharedRealm realm, std::string role, size_t count) {
+        CppContext c(realm);
+        for (size_t i = 0; i < count; ++i) {
+            auto obj = Object::create(c, realm, "Person",
+                                      std::any(AnyDict{
+                                          {"_id", ObjectId::gen()},
+                                          {"role", role},
+                                          {"firstName", util::format("%1-%2", role, i)},
+                                          {"lastName", util::format("last-name-%1", i)},
+                                      }));
+        }
+    };
+
+    enum BootstrapMode { NoReconnect, None, SingleMessage, SingleMessageMulti, MultiMessage, Any };
+
+    struct TestParams {
+        size_t num_emps = 500;
+        size_t num_mgrs = 10;
+        size_t num_dirs = 5;
+        std::optional<size_t> num_objects = std::nullopt;
+    };
+
+    struct ExpectedResults {
+        BootstrapMode bootstrap;
+        size_t emps;
+        size_t mgrs;
+        size_t dirs;
+    };
+
+    enum TestState {
+        not_ready,
+        start,
+        reconnect_received,
+        session_resumed,
+        ident_message,
+        downloading,
+        downloaded,
+        complete
+    };
+    TestingStateMachine<TestState> state_machina(TestState::not_ready);
+    int64_t query_version = 0;
+    bool did_client_reset = false;
+    BootstrapMode bootstrap_mode = BootstrapMode::None;
+    size_t download_msg_count = 0;
+    size_t bootstrap_msg_count = 0;
+    bool role_change_bootstrap = false;
+    bool send_test_command = false;
+    auto logger = util::Logger::get_default_logger();
+
+    auto setup_harness = [&](FLXSyncTestHarness& harness, TestParams params) {
+        auto& app_session = harness.session().app_session();
+        /** TODO: Remove when switching to use Protocol version in RCORE-1972 */
```

Comment:
you're doing it as part of this pr.

Reply:
But the server hasn't been updated to use the protocol version yet... I believe it is still behind the feature flag

Reply:
Yes - it is just controlled by a feature flag